R/apa_print_emm_lsm.R

```diff
@@ -68,143 +69,186 @@ apa_print.summary_emm <- function(
   #   return(apa_res)
   # }
 
-  ci <- get_emm_conf_level(x)
-  ci_supplied <- !length(ci) == 0
+  null.value <- x$null.value
+  if(is.null(null.value)) {
+    null.value <- 0
+  }
+  # delta <- attr(x, "delta")
+  # one.sided <- attr(x, "side")
+
+  tidy_x <- data.frame(broom::tidy(x, null.value = null.value))
+
+  # Hot fix while we wait on a merge of this PR: https://github.com/tidymodels/broom/pull/1047
+  tidy_x <- rename_column(
+    tidy_x
+    , c("prob", "rate", "ratio", "odds.ratio", "asymp.LCL", "asymp.UCL", "z.ratio")
+    , c("estimate", "estimate", "estimate", "estimate", "conf.low", "conf.high", "statistic")
+  )
+
+  conf_level <- get_emm_conf_level(x)
+  ci_supplied <- !length(conf_level) == 0
   p_supplied <- "p.value" %in% colnames(x)
-  if(!ci_supplied & !p_supplied) stop("Object 'x' includes neither confidence intervals nor test statistics (i.e., p-values). See '?lsmeans::summary' for details.")
+  # if(!ci_supplied & !p_supplied) stop("Object 'x' includes neither confidence intervals nor test statistics (i.e., p-values). See '?lsmeans::summary' for details.")
 
   if(!ci_supplied) {
-    warning("Object 'x' does not include confidence intervals. APA guidelines recommend to routinely report confidence intervals for all estimates.")
+    if(p_supplied) {
+      warning("Object 'x' does not include confidence intervals. APA guidelines recommend to routinely report confidence intervals for all estimates.")
+    }
 
     conf_level <- NULL
   } else {
-    if(ci < 1) ci <- ci * 100
-    conf_level <- paste0(ci, "\\% CI")
+    if(conf_level < 1) conf_level <- conf_level * 100
+    conf_level <- paste0(conf_level, "\\% CI")
   }
 
   if(!p_supplied) {
-    warning("Object 'x' does not include test statistics (i.e., p-values).")
+    # warning("Object 'x' does not include test statistics (i.e., p-values).")
 
     df_colname <- NULL
     stat_colnames <- NULL
   } else {
-    dfdigits <- as.numeric(x$df %%1 > 0) * 2
+    df_colname <- names(tidy_x)[grepl("df\\.*", names(tidy_x))]
+    dfdigits <- as.numeric(x[[df_colname]] %%1 > 0) * 2
```

Comment:
This would be a use case for `print_df()`, but I think it will be better to switch to `beautify()`, which in turn applies `print_df()`